app/service_objects/cocktail_counter.rb

```diff
@@ -0,0 +1,18 @@
+class CocktailCounter
+
+  # @param cocktail: Recipe
+  #
+  # Given a cocktail (or recipe), returns how many the user can make
+  # with their reagents.
+  def self.count_possible_cocktails(cocktail, user)
+    cocktail.matching_reagents(user).map do |amount, reagents|
```

Comment:
Looking at this more deeply, for premature optimization reasons I do want to do this query more efficiently. At the very least we should pull all reagents for every favorite cocktail in one query. I'll look into that